### PR TITLE
[bugfix] fix hgraph concurrent build & continue add problem

### PR DESCRIPTION
**Pull request by LHT129**
fixed: #208

### CODE REVIEW

File: src/data_cell/graph_datacell.h [hunk 1]
```diff
@@ -138,6 +141,18 @@ GraphDataCell<IOTmpl, false>::GetNeighbors(InnerIdType id,
...
 
+template <typename IOTmpl>
+void
+GraphDataCell<IOTmpl, false>::Resize(InnerIdType new_size) {
+    if (new_size < this->max_capacity_) {
+        return;
+    }
+    this->max_capacity_ = new_size;
+    uint64_t io_size = new_size * code_line_size_;
+    uint8_t end_flag = 127;
```
Comment (inabao) [inline src/data_cell/graph_datacell.h]: Defined as static and constant.

Comment (LHT129) [reply src/data_cell/graph_datacell.h]: actually the value of this is meaningless, I think add a comment is useful

File: src/index/hgraph_zparameters.cpp [hunk 2]
```diff
@@ -131,7 +131,7 @@ const std::string HGraphParameters::DEFAULT_HGRAPH_PARAMS = format_map(
         },
         "{BUILD_PARAMS_KEY}": {
             "{BUILD_EF_CONSTRUCTION}": 400,
-            "{BUILD_THREAD_COUNT}": 5
+            "{BUILD_THREAD_COUNT}": 100
```
Comment (jiaweizone) [inline src/index/hgraph_zparameters.cpp:134]: why change to 100 ?

File: src/default_allocator.cpp [hunk 3]
```diff
@@ -34,7 +34,7 @@ DefaultAllocator::Deallocate(void* p) {
     if (!p) {
         return;
     }
-    std::lock_guard<std::mutex> guard(set_mutex_);
+    std::unique_lock<std::mutex> guard(set_mutex_);
```
Comment (jiaweizone) [inline src/default_allocator.cpp]: Why change to `unique_lock` ? here no need manual lock or unlock

File: src/index/hgraph_zparameters.cpp (hunk 2 above)
Comment (LHT129) [reply src/index/hgraph_zparameters.cpp:134]: the same as the default value in hgraph index, also easy to test on concurrent

File: src/default_allocator.cpp (hunk 3 above)
Comment (LHT129) [reply src/default_allocator.cpp]: done

**Review by inabao — APPROVED** [review]
LGTM

File: src/algorithm/hgraph.cpp [hunk 4]
```diff
@@ -775,6 +775,8 @@ HGraph::add_one_point(const float* data, int level, InnerIdType inner_id) {
         .is_id_allowed_ = nullptr,
     };
 
+    std::unique_lock cur_lock(this->neighbors_mutex_[inner_id]);
```
Comment (jiaweizone) [inline src/algorithm/hgraph.cpp]: ditto

Comment (LHT129) [reply src/algorithm/hgraph.cpp]: done

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by ShawnShawnYou — APPROVED** [review]

**Pull request merged**